examples/tutorial/tutorial_notebooks/introduction/exercise_4.ipynb

```diff
@@ -0,0 +1,159 @@
+   "metadata": {},
+   "source": [
+    "### Concept for exercise: Progress Bar\n",
+    "\n",
+    "\n",
+    "Sometimes when running long functions on DataFrames, it can be hard to tell how much progress has been made, as well as how much longer the function will run. A progress bar allows users to see the estimated progress and completion time of each line they run, in environments such as a shell or Jupyter notebook.\n",
+    "\n",
+    "To enable Modin's Progress Bar, add the following lines of code after importing `modin.pandas`:\n",
+    "```python\n",
+    "from tqdm import tqdm\n",
+    "from modin.config import ProgressBar\n",
+    "ProgressBar.enable()\n",
+    "```\n",
+    "\n",
+    "In this exercise, we'll see how the progress bar can improve our experience running dataframe queries!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f95d4874",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "import modin.pandas as pd\n",
+    "import numpy as np\n",
+    "from tqdm import tqdm\n",
+    "from modin.config import ProgressBar\n",
+    "ProgressBar.enable()\n",
+    "\n",
+    "frame_data = np.random.randint(0, 100, size=(2**18, 2**8))\n",
+    "df = pd.DataFrame(frame_data).add_prefix(\"col\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6905bc6b",
+   "metadata": {},
+   "source": [
+    "On longer functions, its nice to be able to see an estimation of how much longer things will take!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "236ec8e2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = df.applymap(lambda x: ~x)\n",
+    "df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e7bf87a5",
+   "metadata": {},
+   "source": [
+    "### Concept for exercise: Spreadsheet\n",
+    "\n",
+    "For those who have worked with Excel, the Spreadsheet API will definitely feel familiar! The Spreadsheet API is a Jupyter notebook widget that allows us to interact with Modin DataFrames in a spreadsheet-like fashion while taking advantage of the underlying capabilities of Modin. The widget makes it quick and easy to explore, sort, filter, and edit data as well as export the changes as reproducible code.\n",
+    "\n",
+    "Let's look back at a subset of the 2015 NYC Taxi Data from Exercise 2, and see how the Spreadsheet API can make it easy to play with the data!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5d5c4a3e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!jupyter nbextension enable --py --sys-prefix modin_spreadsheet\n",
+    "ProgressBar.disable()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dc8d5903",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import modin.experimental.spreadsheet as mss\n",
+    "\n",
+    "s3_path = \"s3://dask-data/nyc-taxi/2015/yellow_tripdata_2015-01.csv\"\n",
```

Comment:
The s3 path in exercise_2 and 4 takes a while to download. Wondering if we should just have users download this once. Similar to what is suggested in the [quickstart guide](https://modin--3898.org.readthedocs.build/en/3898/getting_started/quickstart.html#example-instant-scalability-with-no-extra-effort): 
```python
# This may take a few minutes to download
import urllib.request
s3_path = "https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2021-01.csv"
urllib.request.urlretrieve(s3_path, "taxi.csv")
```

Reply:
That might be good @dorisjlee, but could we make it optional?

Reply:
Added and made this optional